packages/embed/src/index.ts

```diff
@@ -38,17 +58,28 @@ export function setup(setupConfig: SetupConfig): void {
   if (config.form && !(config.form instanceof Element)) {
     config.form = document.querySelector(config.form) as HTMLFormElement
   }
-  const channel: string = generateChannelId()
-  const iframeUrl: URL = getFrameUrl({ channel, config })
+
   const subjectManager = createSubjectManager()
 
-  // Remove existing elements from the DOM
+  const existingEmbedId = (config.element as HTMLElement).dataset.embedId
+
+  // Cleanup existing element
   if (config.element.hasChildNodes()) {
     while (config.element.firstChild) {
       config.element.removeChild(config.element.lastChild)
     }
   }
 
+  // Cleanup existing events
+  if (cleanup.has(existingEmbedId)) {
```

Comment:
This tags the element with a id for that instance - This mans we can keep a map of cleanup functions.